addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -678,16 +678,47 @@ var FileWidget = SearchableMediaWidget.extend({
     /**
      * @private
      */
-    _onURLInputChange: function () {
+    _onURLInputChange: async function() {
         const inputValue = this.$urlInput.val().split('?')[0];
+        const domainExtensions = [
```

Comment:
you know there is a lot more top level domain than that?
https://en.wikipedia.org/wiki/List_of_Internet_top-level_domains
also an URL can be internationalized and contains unicode

Reply:
@mart-e should those domains to be added here as well?, i simply used the domain list we already have in utils https://github.com/odoo/odoo/blob/master/addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js#L26.

Reply:
@shsa-odoo checking the extension does not sound like a good heuristic to detect domain names. There are many regex to do so on stackoverflow, some more advanced than others based on to which level you want and what you want to exclude.

Reply:
@dmo-odoo , @mart-e so what should we do here, shall we carryon with tldWhitelist in our case its `domainExtensions` already used in [utils](https://github.com/odoo/odoo/blob/master/addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js#L26). Or we should write any generalized one ? I think @deso-odoo is also stuck with the same scenario in this [task](https://www.odoo.com/web#id=3468646&menu_id=4720&cids=2&action=333&active_id=1695&model=project.task&view_type=form) in which he is stuck which approach to be followed whether to keep `tldWhitelist` as base for domains which doesnot contain all the domains.
If we keep the current `tldWhitelist` logic we miss out many other TLD's.
and if we try something like `[-a-zA-Z0-9@:%._\+~#=]{1,64}` to test domains it will consider many as domains which are not domains in actual terms.

Reply:
Wait, I'm confused. Why are we validating url and whatnot here ? Wasn't this task supposed to be about making HEAD a request to see if it's an image and that's it ?